src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1011,6 +1023,73 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
 
             break;
         }
+        case WM_TOUCH:
+        {
+            // Get the number of events
+            std::int16_t num = LOWORD(wParam);
+
+            // Reserve memory
+            std::unique_ptr<TOUCHINPUT[]> events = std::make_unique<TOUCHINPUT[]>(num);
+
+            if (events)
+            {
```

Comment:
`if (events)` will always be true. Allocation failure is signalled through `std::bad_alloc`.

I honestly don't really see the point of using unique pointers for dynamic arrays. Yes, it's possible, but also everyone will ask "why not `std::vector`"? It _may_ save a few elements of allocation, but I'd argue clarity is more useful, especially as you have a proper container API.

I would probably use something like:
```cpp
std::vector<TOUCHINPUT> events(num);

TOUCHINPUT event;
if (GetTouchInputInfo(reinterpret_cast<HTOUCHINPUT>(lParam), num, events.data(), sizeof(TOUCHINPUT)) != 0)
{
    // error handling + break or return
}

for (TOUCHINPUT& event : events)
{
    ...
}
```